### PR TITLE
Refactored Rosetta tests - separated construction tests

### PR DESCRIPTION
**Pull request by MHassanTariq**
## Description

This PR separates the rosetta tests from construction tests in `api.ts`. I have also refactored and removed integration for other tests. However, construction tests require integration with `EventServer`, therefore, those are not refactored at all. 

For details refer to issue #1077

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [X] Other

## Checklist
- [ ] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [ ] `npm run test` passes
- [ ] Changelog is updated
- [ ] Tag 1 of @rafaelcr or @zone117x for review

### CODE REVIEW

**Comment by rafaelcr** [comment]
@MHassanTariq this looks good, but were you able to find the cause for the flakyness? If the construction tests were not refactored, do we still have a risk of flakyness in those?

**Comment by MHassanTariq** [comment]
@rafaelcr I initially spent a lot of time diagnosing the actual problem. However, I wasn't able to pinpoint the exact cause as they were not breaking regularly. Therefore, I refactored the tests to the extent where I could remove the maximum integration between them.

**Comment by rafaelcr** [comment]
Great news @MHassanTariq ... I think the strategy then would be to follow @CharlieC3 's advice and add a mechanism to Github actions that re-tries the test suite if it detects that same error. Can you add that to this PR?

**Comment by CharlieC3** [comment]
> Great news @MHassanTariq ... I think the strategy then would be to follow @CharlieC3 's advice and add a mechanism to Github actions that re-tries the test suite if it detects that same error. Can you add that to this PR?

I think the long-term goal would be to use a custom GH runner, or figure out a way to change how the tests are ran to consume less memory. Otherwise besides the retries, we could try running **just these tests** (not the whole pipeline) on a MacOS GH runner. We get fewer of those GH runners, [but they offer double the amount of memory](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources).

**Comment by MHassanTariq** [comment]
@rafaelcr sure. I'll look into it.
@CharlieC3 thanks for the suggestion and link!

**Comment by MHassanTariq** [comment]
@CharlieC3 and @rafaelcr I've tried re-running the tests for `rosetta` once. I think it fixes the problem temporarily like @CharlieC3 mentioned. 
Can you please go through the [changes](https://github.com/hirosystems/stacks-blockchain-api/pull/1103/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR253-R255) in `ci.yml` and let me know if it looks good?

File: src/tests-rosetta/construction.ts [hunk 1]
```diff
@@ -0,0 +1,2492 @@
...
+        tries++;
+        await  new StacksCoreRpcClient().getPox();
+        return Promise.resolve();
+      }
+      catch(error){
+        console.log('Error getting pox info on try ' + tries, error);
+        timeout(100);
+      }
+    }
+  }
```
Comment (rafaelcr) [inline src/tests-rosetta/construction.ts:109]: Echoing what we discussed on the call, this function could be improved to wait for PoX in a robust way

**Review by rafaelcr — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -250,6 +250,10 @@ jobs:
       - name: Run tests
         run: npm run test:rosetta
 
+      - name: Re-run tests
+        run: npm run test:rosetta
+        if: failure()
```
Comment (rafaelcr) [inline .github/workflows/ci.yml:255]: This will not be necessary anymore

**Review by rafaelcr — APPROVED** [review]

**Pull request merged**